pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -164,10 +181,12 @@ func (s *Service) SendExchangeResponse(exchangeResponse *Response, destination s
 }
 
 func (s *Service) marshalAndSend(data interface{}, errorMsg, destination string) (string, error) {
+	// TODO need access to the wallet in order to first pack() the msg before sending
```

Comment:
It would be good to have an outbound sender function injected, and that function takes care of calling pack.

Reply:
As we just discussed, we also need a way to map a URI to an outbound transport.

i.e., an outbound dispatcher that does both the mapping and the packing.